src/NRules/NRules.RuleModel/Builders/CollectPatternBuilder.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+
+namespace NRules.RuleModel.Builders
+{
+    /// <summary>
+    /// Builder to compose a Collect rule pattern.
+    /// </summary>
+    public class CollectPatternBuilder : PatternBuilder
```

Comment:
Instead of a `CollectPatternBuilder`, I would handle this fully inside `QueryExpression`.
Instead of returning a `PatternBuilder` from the `_buildAction`, it should return a new type, something like:
```c#
private class AggregateBuildResult
{
    public PatternBuilder Pattern { get; }
    public AggregateBuilder Aggregate { get; }
    public PatternBuilder Source { get; }
}
```
Then you can manipulate the `AggregateBuilder` inside the `OrderBy` to call `Sort` on it.